agents/agent3.py

```diff
@@ -942,6 +940,7 @@ def __init__(self, dictionary: Optional[List[str]] = None) -> None:
             self.codec = HuffmanCodec.from_frequencies(dictionary)
         else:
             self.codec = load_shakespeare()
+        self.modified_codec = {symbol: bin(val)[2:].rjust(bits, '0') for symbol, (bits, val) in self.codec.get_code_table().items()}
```

Comment:
how is this different from the old codec? I might have missed something.

Reply:
dependent on vocabulary size.

dahuffman might insert a padding bit for all chars... so we are removing it here.